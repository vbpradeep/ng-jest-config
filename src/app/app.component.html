<h1>Hello, {{ title }}</h1>

<app-api-component></app-api-component>

<app-event-component></app-event-component>


<router-outlet></router-outlet>