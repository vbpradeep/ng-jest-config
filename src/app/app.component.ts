import { Component } from '@angular/core';
import { RouterOutlet } from '@angular/router';
import { ApiComponentComponent } from '../components/API-component/api-component/api-component.component';
import { EventComponentComponent } from '../components/Event-component/event-component/event-component.component';
import {CONSTANTS } from '@constants';

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [RouterOutlet, ApiComponentComponent, EventComponentComponent],
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss'
})
export class AppComponent {
  title = 'ng-jest-integration';

  constructor() {
  }

  ngOnInit() {
    this.title = CONSTANTS;
  }
}
