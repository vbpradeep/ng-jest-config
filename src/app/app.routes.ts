import { Routes } from '@angular/router';
import { AppComponent } from './app.component';
import { APIResolver } from '../services/api.resolver';
import { ResolverComponentComponent } from '../components/Resolver-component/resolver-component/resolver-component.component';

export const routes: Routes = [
    {
        path: 'data',
        component: ResolverComponentComponent,
        resolve: { data: APIResolver },
      }
];
