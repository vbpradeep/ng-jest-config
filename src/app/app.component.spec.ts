import { ComponentFixture, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { HttpTestingController, provideHttpClientTesting } from '@angular/common/http/testing';
import { provideHttpClient } from '@angular/common/http';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [ provideHttpClient(),
        provideHttpClientTesting(),]
    }).compileComponents();
    TestBed.inject(HttpTestingController);
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it(`should have as title 'ng-jest-integration'`, () => {
    expect(component.title).toEqual('ng-jest-integration');
  });

  it(`should have as title 'ng-jest-integration'`, () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
