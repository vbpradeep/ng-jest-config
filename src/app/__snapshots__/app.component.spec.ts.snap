// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should have as title 'ng-jest-integration' 1`] = `
<div
  id="root2"
>
  <h1>
    Hello, ng-jest-integration
  </h1>
  <app-api-component>
    
  </app-api-component>
  <app-event-component>
    <p>
      event-component works!
    </p>
    <button
      id="btn"
    >
       Click! 
    </button>
  </app-event-component>
  <router-outlet />
  
</div>
`;
